pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -32,6 +32,7 @@ type NodeTree struct {
 	tree      map[string]*nodeArray // a map from zone (region-zone) to an array of nodes in the zone.
 	zones     []string              // a list of all the zones in the tree (keys)
 	zoneIndex int
+	allNodes  []string
```

Comment:
I'm thinking about whether we can store the node list in a [sync.Value](https://golang.org/pkg/sync/atomic/#Value), as this array is read frequently(inside `AllNodes` method) and updated infrequently(only when node added/removed), just like this [golang example](https://golang.org/pkg/sync/atomic/#example_Value_readMostly).

In fact, I don't know how much this can help, but at least we can test it as a candidate.

Reply:
@mlmhl I haven't looked into the implementation of sync.Value. It may not be much more efficient than acquiring/releasing a lock.